src/Css/Stylesheet.php

```diff
@@ -649,6 +649,18 @@ protected function selectorToXpath(string $selector, bool $firstPass = false): ?
                             $query .= "[" . implode(" or ", $elements) . "]";
                             break;
 
+                        case "not":
+                            $p = $i + 1;
+                            $length = strpos($selector, ")", $i) - $p;
+                            $i += $length + 1;
+                            $matchList = trim(mb_substr($selector, $p, $length));
+                            $subquery = $this->selectorToXpath($matchList, true)['query'];
+                            if(substr($subquery, 0, 4) != '//*[') {
+                                return null;		//not supported
+                            }
+                            $query .= "[not(".substr($subquery, 4, -1).")]";
```

Comment:
Recursing into selectorToXpath is problematic since we don't want a full XPath query. That being said the following small tweak seems to work better, though it's still not great. It supports both elements and attributes, but combined selectors (e.g., `element.class` or `.class1.class2`).

```suggestion
                            $subqueries = array_map("trim", explode(",", $matchList));
                            foreach ($subqueries as &$subquery) {
                                if (preg_match("/^\w/", $subquery)) {
                                    $subquery = "name() = '$subquery'";
                                } else {
                                    $subquery = substr($this->selectorToXpath($subquery, true)['query'], 4, -1);
                                }
                            }
                            $query .= "[not(" . implode(" or ", $subqueries) . ")]";
```